modules/lightning_features/lightning_api/modules/api_test/tests/src/Functional/EntityCRUTest.php

```diff
@@ -0,0 +1,178 @@
+    $this->assertEquals($description, $body['data']['attributes']['description']);
+
+    $new_description = 'Refined by Johann Bartsch.';
+    $data = [
+      'data' => [
+        'id' => 'taxonomy_test_vocabulary',
+        'attributes' => [
+          'description' => $new_description,
+        ]
+      ]
+    ];
+
+    // Update the vocabulary.
+    $this->request('/jsonapi/taxonomy_vocabulary/taxonomy_vocabulary/taxonomy_test_vocabulary', 'patch', $this->token, $data);
+
+    // Read the updated vocabulary.
+    $response = $this->request('/jsonapi/taxonomy_vocabulary/taxonomy_vocabulary/taxonomy_test_vocabulary', 'get', $this->token);
+    $body = $this->decodeResponse($response);
+    $this->assertEquals('I\'m a vocab', $body['data']['attributes']['name']);
+    $this->assertEquals($new_description, $body['data']['attributes']['description']);
+
+    // Assert that the newly created vocabulary's endpoint is reachable.
+    // @todo figure out why we need to rebuild caches for it to be available.
+    drupal_flush_all_caches();
+    $response = $this->request('/jsonapi/taxonomy_term/im_a_vocab');
+    $this->assertEquals(200, $response->getStatusCode());
+
+    $description = 'How quickly deft jumping zebras vex.';
+    $data = [
+      'data' => [
+        'type' => 'taxonomy_term--im_a_vocab',
+        'id' => 'zebra_taxonomy_term',
+        'attributes' => [
+          'name' => 'zebra',
+          'uuid' => 'zebra_taxonomy_term',
+          'description' => [
+            'value' => $description,
+            'format' => 'rich_text',
+          ]
+        ],
+        'relationships' => [
+          'vid' => [
+            'data' => [
+              'type' => 'taxonomy_vocabulary--taxonomy_vocabulary',
+              'id' => 'taxonomy_test_vocabulary',
+            ]
+          ]
+        ]
+      ]
+    ];
+
+    // Create a taxonomy term (content entity).
+    $this->request('/jsonapi/taxonomy_term/im_a_vocab', 'post', $this->token, $data);
+
+    // Read the taxonomy term.
+    $response = $this->request('/jsonapi/taxonomy_term/im_a_vocab/zebra_taxonomy_term', 'get', $this->token);
+    $body = $this->decodeResponse($response);
+    $this->assertEquals($description, $body['data']['attributes']['description']['value']);
+
+    $new_description = 'Smart squid gives lazy lummox who asks for job pen.';
+    $data = [
+      'data' => [
+        'id' => 'zebra_taxonomy_term',
+        'attributes' => [
+          'description' => [
+            'value' => $new_description,
+          ]
+        ]
+      ]
+    ];
+
+    // Update the taxonomy term.
+    $this->request('/jsonapi/taxonomy_term/im_a_vocab/zebra_taxonomy_term', 'patch', $this->token, $data);
+
+    // Read the updated taxonomy term.
+    $response = $this->request('/jsonapi/taxonomy_term/im_a_vocab/zebra_taxonomy_term', 'get', $this->token);
+    $body = $this->decodeResponse($response);
+    $this->assertEquals('zebra', $body['data']['attributes']['name']);
```

Comment:
We should use assertSame() when comparing strings, because of PHP's weird string handling quirks.

Reply:
✔️